### PR TITLE
chore: add package entrypoints enforcement script

### PR DESCRIPTION
**Pull request by kuhe**
paired with https://github.com/awslabs/smithy-typescript/pull/1066 (see explanation there)

### CODE REVIEW

**Review by syall — APPROVED** [review]

File: packages/credential-provider-web-identity/package.json [hunk 1]
```diff
@@ -15,7 +15,6 @@
     "test": "jest"
   },
   "browser": {
-    "./dist-cjs/fromTokenFile": false,
```
Comment (trivikr) [inline packages/credential-provider-web-identity/package.json:18]: Make script more restrictive to check if the value is a string, and ignore if it's boolean?

This is needed as for webpack support https://github.com/aws/aws-sdk-js-v3/pull/3005

File: packages/rds-signer/package.json [hunk 2]
```diff
@@ -60,9 +60,7 @@
   "browser": {
     "./dist-es/runtimeConfig": "./dist-es/runtimeConfig.browser"
   },
-  "react-native": {
-    "./dist-es/runtimeConfig": "./dist-es/runtimeConfig.native"
```
Comment (trivikr) [inline packages/rds-signer/package.json:64]: Is this removal on purpose? Does React Native has any other bundlers which might require this configuration?

If not, can we remove the `react-native` key itself?

File: packages/credential-provider-web-identity/package.json (hunk 1 above)
Comment (kuhe) [reply packages/credential-provider-web-identity/package.json:18]: we don't provide dist-cjs corrections in other packages for the browser environment

File: packages/rds-signer/package.json (hunk 2 above)
Comment (kuhe) [reply packages/rds-signer/package.json:64]: this removal is on purpose because you asked that we do not explicitly add the replacements that would be automatic. 

react-native is left as an object because of the consistency enforcement that `browser` and `react-native` are of the same type: object or string or undefined.

this prevents errors like forgetting to add the react-native directive when creating a browser replacement

File: packages/credential-provider-web-identity/package.json (hunk 1 above)
Comment (kuhe) [reply packages/credential-provider-web-identity/package.json:18]: added condition to skip dropping boolean directives in browser with CJS to avoid unforeseen consequences

**Review by trivikr — APPROVED** [review]
